src/components/Send/SendScreen.js

```diff
@@ -227,6 +229,7 @@ class SendScreen extends Component<Props, State> {
       this.handleAmountChange(CONFIG.DEBUG.SEND_AMOUNT)
     }
     this.props.navigation.setParams({onScanAddress: this.handleAddressChange})
+    FlagSecure.activate()
```

Comment:
I think this way may lead to a lot of bugs. As I understood `componentDidMount` is `is invoked immediately after a component is mounted (inserted into the tree).` If the component is not removed from the stack and another screen is pushed, it will not deactivate it. We basically want to activate/ deactivate screens based if they are been shown, rather than when they were loaded (sometimes is the same).

I think we want to resemble something similar to `viewWillAppear` or `viewDidAppear` from iOS.

Reply:
You are right! So I did further research on this and it seems we might achieve what you suggest through the `ReactNavigation` API: https://reactnavigation.org/docs/en/navigation-events.html

What do you think of this approach?

Reply:
I think it makes sense! I couldn’t verify what `willFocus` and `didFocus` do / behave. The links seem broken https://facebook.github.io/react-native/docs/0.43/navigator.md#onwillfocus 🤔